ignore alternative authorization header prefixes (#4904)
apollographql/router: .changesets/feat_lleadbet_ignore_other_auth_prefixes.md

### Ignore other auth prefixes in the JWT plugin

> If the router encounters an authorization header with a different prefix in the value than what it expects, it will now ignore it. If the router was configured without the `require_authentication` option or without the authorization directives, then some requests that came with a different header prefix that were rejected before will now go through the router. If those options were configured, then there will be no change in behaviour.

Reviewer: Attempted clarification, but make sure it still makes sense.

```suggestion
If the router encounters an authorization header with a different prefix to the configured value, it will now ignore it.

If the router is configured without the `require_authentication` option or without the authorization directives, then requests that have a different header prefix, that were previously rejected, will now go through the router.

If both of those options were configured, then there will be no change in behaviour.
```

> As an example, with a router configure like this:

Reviewer: ```suggestion
As an example, with a router configured like this:
```

```yaml title="router.yaml"
authentication:
  router:
    jwt:
      header_name: authorization
      header_value_prefix: "Bearer"
```

> In the above, the router will ignore `Authorization: Basic <token>`, but process requests with `Authorization: Bearer <token>` defined.

Reviewer: ```suggestion
In the above, the router will ignore requests with a `Authorization: Basic <token>` header and process requests with a `Authorization: Bearer <token>` header.
```
